staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -404,6 +405,11 @@ redirectLoop:
 		if err != nil {
 			return nil, nil, fmt.Errorf("malformed Location header: %v", err)
 		}
+
+		// Only allow redirects to the same host.
+		if validateRedirects && location.Hostname() != originalLocation.Hostname() {
```

Comment:
I can definitely imagine services that notice when they're not accessed by hostname and redirect to fix that.

Reply:
This is only used for serving exec/attach/port-forward requests, when the kubelet redirects to the CRI streaming server. AFAIK all the CRI implementations use the [same library](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/server/streaming/server.go) to implement that server. The main case I'm worried about is when the CRI is configured to return a URL to a different interface than the one the kubelet used.

Reply:
hmm, actually I guess that could matter if using portforward, and the destination is a redirect to an external endpoint. Maybe rather than returning an error in this case, the apiserver should just forward the redirect back to the client?